examples/allocator-service/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"agones.dev/agones/pkg/apis/stable/v1alpha1"
+	"agones.dev/agones/pkg/client/clientset/versioned"
+	"agones.dev/agones/pkg/util/runtime" // for the logger
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+const namespace = "default"
+const fleetname = "simple-udp"
+const generatename = "simple-udp-"
+
+var (
+	logger  = runtime.NewLoggerWithSource("main")
+	address = "NoReadyGS" // default response if no gameservers are available
+	port    int32         // 0 until populated
+)
+
+// A handler for the web server
+type handler func(w http.ResponseWriter, r *http.Request)
+
+// The structure of the json response
+type result struct {
+	IPAddress string `json:"address"`
+	Port      int32  `json:"port"`
+}
+
+// Move a replica from ready to allocated and return the ip and port
+func getIPAddress() (string, int32) {
+	// Create the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.WithError(err).Fatal("Could not create in cluster config")
+	}
+
+	// Access to the Agones resources through the Agones Clientset
+	agonesClient, err := versioned.NewForConfig(config)
+	if err != nil {
+		logger.WithError(err).Fatal("Could not create the agones api clientset")
+	} else {
+		logger.Info("Created the agones api clientset")
+	}
+
+	// Log the values used in the fleet allocation
+	logger.WithField("namespace", namespace).Info("namespace for fa")
+	logger.WithField("generatename", generatename).Info("generatename for fa")
+	logger.WithField("fleetname", fleetname).Info("fleetname for fa")
+
+	// Get a FleetAllocationInterface for this namespace
+	fleetAllocationInterface := agonesClient.StableV1alpha1().FleetAllocations(namespace)
+
+	// Define the fleet allocation
+	fa := &v1alpha1.FleetAllocation{
+		ObjectMeta: v1.ObjectMeta{
+			GenerateName: generatename, Namespace: namespace,
+		},
+		Spec: v1alpha1.FleetAllocationSpec{FleetName: fleetname},
+	}
+
+	// Create a new fleet allocation
+	newFleetAllocation, err := fleetAllocationInterface.Create(fa)
+	if err != nil {
```

Comment:
Yes, definitely should return an error.  Previously I was checking to see if there was a Ready gameserver and if not spawning a new one.  The defaults for IP and port would be returned if nothing was available.
```
	address = "NoReadyGS" // default response if no gameservers are available
	port    int32         // 0 until populated
```
But I think in removing my checks for a ready server and the spawning of a new one I might have changed things so that the return of defaults is no longer happening on error.  I'll add that check back and differentiate between the fleetallocation failure scenario and the ready gameserver unavailable scenario.  Then the return value can make sense for those exceptions.

How about 500 Internal Server Error as a return code?  If I were coding against this and got a 404 I would think the enpdoint was not being found.

Reply:
500 makes sense to me. :+1: